cmd/completion/complete_functions.go

```diff
@@ -0,0 +1,233 @@
+package completion
+
+import (
+	"github.com/cloudfoundry/bosh-cli/v7/cmd/opts"
+	boshlog "github.com/cloudfoundry/bosh-utils/logger"
+	"github.com/spf13/cobra"
+	"reflect"
+)
+
+type CompleteFunctions struct {
+	directorQuery DirectorQueryInterface
+	logger        boshlog.Logger
+	logTag        string
+}
+type CompleteFunctionsMap map[string]func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective)
+
+func NewCompleteFunctionsMap(logger boshlog.Logger, directorQuery DirectorQueryInterface) *CompleteFunctionsMap {
+	c := &CompleteFunctions{
+		directorQuery: directorQuery,
+		logger:        logger,
+		logTag:        "completion.CompleteFunctions",
+	}
+	cfm := &CompleteFunctionsMap{
+		"--deployment": c.listDeploymentNames,
+		"--dir":        c.listDirectories,
```

Comment:
Is it possible for this string, and the others below, to come from [opts.BoshOpts.DeploymentOpt](cmd/opts/opts.go:27) rather than being hard-coded here?

Reply:
I tried but I don't know how to do it in an elegant way. Maybe these literals should be declared as constants?

Reply:
Agreed that this isn't easy. I think we can look at figuring this out once the PR is merged.